src/main/java/com/hadoop/compression/lzo/DistributedLzoIndexer.java

```diff
@@ -25,57 +27,108 @@
+   */
   private final PathFilter nonTemporaryFilter = new PathFilter() {
+    @Override
     public boolean accept(Path path) {
-      return !path.toString().endsWith("/_temporary");
+      return !path.toString().endsWith(TEMP_FILE_EXTENSION);
     }
   };
 
-  private void walkPath(Path path, PathFilter pathFilter, List<Path> accumulator) {
+  /**
+   * Returns whether a file should be considered small enough to skip indexing.
+   */
+  private boolean isSmallFile(FileStatus status) {
+    return status.getLen() < lzoSmallFileSize;
+  }
+  
+  private void visitPath(Path path, PathFilter pathFilter, List<Path> accumulator, boolean recursive) {
     try {
-      FileSystem fs = path.getFileSystem(getConf());
+      FileSystem fs = path.getFileSystem(this.conf);
       FileStatus fileStatus = fs.getFileStatus(path);
 
-      if (fileStatus.isDir()) {
-        FileStatus[] children = fs.listStatus(path, pathFilter);
-        for (FileStatus childStatus : children) {
-          walkPath(childStatus.getPath(), pathFilter, accumulator);
-        }
-      } else if (path.toString().endsWith(LZO_EXTENSION)) {
-        Path lzoIndexPath = path.suffix(LzoIndex.LZO_INDEX_SUFFIX);
-        if (fs.exists(lzoIndexPath)) {
-          // If the index exists and is of nonzero size, we're already done.
-          // We re-index a file with a zero-length index, because every file has at least one block.
-          if (fs.getFileStatus(lzoIndexPath).getLen() > 0) {
-            LOG.info("[SKIP] LZO index file already exists for " + path);
-            return;
-          } else {
-            LOG.info("Adding LZO file " + path + " to indexing list (index file exists but is zero length)");
-            accumulator.add(path);
+      if (fileStatus.isDirectory()) {
+        if (recursive) {
+          FileStatus[] children = fs.listStatus(path, pathFilter);
+          for (FileStatus childStatus : children) {
+            visitPath(childStatus.getPath(), pathFilter, accumulator, recursive);
           }
         } else {
-          // If no index exists, we need to index the file.
-          LOG.info("Adding LZO file " + path + " to indexing list (no index currently exists)");
-          accumulator.add(path);
+          LOG.info("[SKIP] Path " + path + " is a directory and recursion is not enabled.");
         }
+      } else if (shouldIndexPath(fileStatus, fs)) {
+        accumulator.add(path);
       }
     } catch (IOException ioe) {
       LOG.warn("Error walking path: " + path, ioe);
     }
   }
 
+  private boolean shouldIndexPath(FileStatus fileStatus, FileSystem fs) throws IOException {
+    Path path = fileStatus.getPath();
+    if (path.toString().endsWith(LZO_EXTENSION)) {
+      if (this.lzoSkipIndexingSmallFiles && isSmallFile(fileStatus)) {
+        LOG.info("[SKIP] Skip indexing small files enabled and " + path + " is too small");
+        return false;
+      }
+
+      Path lzoIndexPath = new Path(path, LzoIndex.LZO_INDEX_SUFFIX);
```

Comment:
I think they're roughly the same. I'm more used to using the constructor. The constructor does some additional resolution of the URIs for parent/child and normalization of the paths. The suffix method does this:
```java
public Path suffix(String suffix) {
    return new Path(this.getParent(), this.getName() + suffix);
}
```
Which does seem like it should also work, so I will switch back to using .suffix here since it is what already works.